From stable-bounces@linux.kernel.org  Tue Apr  7 02:43:35 2009
From: Eric Anholt <eric@anholt.net>
To: stable@kernel.org
Date: Mon,  6 Apr 2009 19:40:22 -0700
Message-Id: <1239072025-1706-11-git-send-email-eric@anholt.net>
Cc: Eric Anholt <eric@anholt.net>, Jesse Barnes <jbarnes@virtuosugeek.org>,         Jesse Barnes <jbarnes@virtuousgeek.org>
Subject: drm/i915: check for -EINVAL from vm_insert_pfn

From: Jesse Barnes <jbarnes@virtuousgeek.org>

upstream commit: 959b887cf42fd63cf10e28a7f26126f78aa1c0b0

Indicates something is wrong with the mapping; and apparently triggers
in current kernels.

Signed-off-by: Jesse Barnes <jbarnes@virtuosugeek.org>
Signed-off-by: Eric Anholt <eric@anholt.net>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---
 drivers/gpu/drm/i915/i915_gem.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -1072,6 +1072,7 @@ int i915_gem_fault(struct vm_area_struct
 	case -EAGAIN:
 		return VM_FAULT_OOM;
 	case -EFAULT:
+	case -EINVAL:
 		return VM_FAULT_SIGBUS;
 	default:
 		return VM_FAULT_NOPAGE;
